zinit-install.zsh

```diff
@@ -1503,22 +1516,31 @@ builtin source "${ZINIT[BIN_DIR]}/zinit-side.zsh" || {
         }
 
         if (( $#list > 1 )) {
-            list2=( ${(M)list[@]:#(#i)*${~matchstr[${${OSTYPE%(#i)-(gnu|musl)}%%(-|)[0-9.]##}]:-${${OSTYPE%(#i)-(gnu|musl)}%%(-|)[0-9.]##}}*} )
+            list2=( ${(M)list[@]:#(#i)*${~matchstr[$MACHTYPE]:-${MACHTYPE#(#i)(i|amd)}}*} )
+            (( $#list2 > 0 )) && list=( ${list2[@]} )
+        }
+
+        if (( ${#list} > 1 && ${#matchstr[${MACHTYPE}-2]} )) {
+            list2=( ${(M)list[@]:#(#i)*${~matchstr[${MACHTYPE}-2]:-${MACHTYPE#(#i)(i|amd)}}*} )
             (( $#list2 > 0 )) && list=( ${list2[@]} )
         }
 
         if (( $#list > 1 )) {
-            list2=( ${list[@]:#(#i)*.(sha[[:digit:]]#|asc)} )
+            list2=( ${(M)list[@]:#(#i)*${~matchstr[$OSTYPE(#i)]:-${OSTYPE#(#i)}}*} )
+
+            +zinit-message "{pre}.zinit-get-latest-gh-r-url-part{msg2} DEBUG:{msg}" \
```

Comment:
Added as debug logging in case someone encounters edge case while testing.

Will be removed.